File: RecipeBook/RecipeBook/Models/Domain/Ingredient.cs
namespace RecipeBook.Models.Domain
{
    public class Ingredient
    {
        public Guid Id { get; set; }
        public IngredientName IngredientName { get; set; }
        public IngredientQuantity IngredientQuantity { get; set; }

        //Navigation Property
        public IEnumerable<Recipe> Recipies { get; set; }
    }
}

File: RecipeBook/client-app/src/app/models/ingredient.ts
export interface Ingredient {
    id: string;
    ingredientName: IngredientName;
    ingredientQuantity: IngredientQuantity;
}

export interface IngredientName {
    id: string;
    name: string;
}

export interface IngredientQuantity {
    id: string;
    quantity: string;
}